lib/widgets/search_box.dart

```diff
@@ -1,67 +1,69 @@
 import "package:flutter/material.dart";
-import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:flutter_hooks/flutter_hooks.dart";
 import "package:flutter_svg/svg.dart";
 
 import "../config/ui_config.dart";
 import "../gen/assets.gen.dart";
 import "../theme/app_theme.dart";
 import "../utils/context_extensions.dart";
 
-class SearchBox extends ConsumerStatefulWidget {
-  final void Function(String query) onQueryChanged;
-  final VoidCallback? onTap;
-  final String? searchText;
-
+class SearchBox extends HookWidget {
   const SearchBox({
     super.key,
     required this.onQueryChanged,
     this.onTap,
     this.searchText,
   });
+  final VoidCallback? onTap;
+  final String? searchText;
+  final void Function(String query) onQueryChanged;
 
   @override
-  ConsumerState<ConsumerStatefulWidget> createState() => _SearchBoxState();
-}
-
-class _SearchBoxState extends ConsumerState<SearchBox> {
-  final controller = TextEditingController();
-  final focusNode = FocusNode();
-  bool showCloseIcon = false;
+  Widget build(BuildContext context) {
+    final focusNode = useFocusNode();
+    final controller = useTextEditingController();
+    final showCloseIcon = useState(false);
 
-  void _onChanged(String query) {
-    setState(() {
-      showCloseIcon = query != "";
-    });
-    widget.onQueryChanged(query);
-  }
+    final onTapOutside = useCallback(
```

Comment:
Follow up to previous question (about `useEffectOnInit`), the callback here'd be executed the first time the hook is invoked, and then whenever focus node changes (meaning the whole list changes)?

Reply:
so there is hook called `useMemoized` which caches some expensive operation/creation of sometiming and recalculates only when someting in the dependecy list changes

Reply:
and `useCallback` is a simple wrapper on `useMemoized` but for functions

Reply:
this basically means it creates this function inside only first time and then on rebuild, it doesn't create this function again but uses previous instance - unless a key in dependecy list has changed

Reply:
> this basically means it creates this function inside only first time and then on rebuild, it doesn't create this function

I thin I got it now! Thank you